skimage/filters/_gabor.py

```diff
@@ -1,48 +1,338 @@
+         \begin{array}{llllll}
+                 x_0     & \quad = r \sin \theta_0 \sin \theta_1 ... \sin \theta_{n-1} \\
+                 x_1     & \quad = r \cos \theta_0 \sin \theta_1 ... \sin \theta_{n-1} \\
+                 x_2     & \quad = r \cos \theta_1 \sin \theta_2 ... \sin \theta_{n-1} \\
+                 ...                                                                   \\
+                 x_{n-1} & \quad = r \cos \theta_{n-2} \sin \theta_{n-1}               \\
+                 x_n     & \quad = r \cos \theta_{n-1}
+         \end{array}
+         \right.
+
+    Parameters
+    ----------
+    r : float
+        Radial coordinate.
+    thetas : (N, ) array
+        Quasipolar angles.
+
+    Returns
+    -------
+    coords : (``N + 1``, ) array
+        Cartesian components of the quasipolar coordinates.
+
+    References
+    ----------
+    .. [1] Tan Mai Nguyen. N-Dimensional Quasipolar Coordinates - Theory and
+           Application. University of Nevada: Las Vegas, Nevada, 2014.
+           https://digitalscholarship.unlv.edu/thesesdissertations/2125
+
+    Notes
+    -----
+    In terms of polar coordinate decomposition:
+
+    .. math::
+
+         \left\{
+         \begin{array}{ll}
+                 y = x_0 = r \sin \theta_0 \\
+                 x = x_1 = r \cos \theta_0
+         \end{array}
+         \right.
+
+    In terms of spherical coordinate decomposition:
+
+    .. math::
+
+         \left\{
+         \begin{array}{lll}
+                 y = x_0 = r \sin \theta_0 \sin \theta_1 \\
+                 x = x_1 = r \cos \theta_0 \sin \theta_1 \\
+                 z = x_2 = r \cos \theta_1
+         \end{array}
+         \right.
+
+    Examples
+    --------
+    >>> _decompose_quasipolar_coords(1, [0])
+    array([ 0.,  1.])
+    >>> _decompose_quasipolar_coords(10, [np.pi / 2, 0])
+    array([  0.,   0.,  10.])
+    """
+    num_axes = len(thetas) + 1
+    coords = r * np.ones(num_axes)
+
+    for which_theta, theta in enumerate(thetas[::-1]):
+        sine = np.sin(theta)
+        theta_index = num_axes - which_theta - 1
+
+        for axis in range(theta_index):
+            coords[axis] *= sine
+
+        coords[theta_index] *= np.cos(theta)
+
+    return coords
+
+
+def _gaussian_kernel(image, center=0, sigma=1, ndim=None):
+    """Multi-dimensional Gaussian kernel.
+
+    Parameters
+    ----------
+    image : non-complex array
+        Linear space to map the filter to.
+    center : scalar or sequence of scalars, optional
+        Center of Gaussian kernel. The coordinates of the center are given for
+        each axis as a sequence, or as a single number, in which case it is
+        equal for all axes.
+    sigma : scalar or sequence of scalars, optional
+        Standard deviation for Gaussian kernel. The standard deviations of the
+        Gaussian filter are given for each axis as a sequence, or as a single
+        number, in which case it is equal for all axes.
+    ndim : int, optional
```

Comment:
I'm wondering here: should we allow center and sigma to be scalars?  If not, we can derive ndim from center and sigma.